### PR TITLE
[FLINK-36757][sql-gateway] Introduce ScripExecutor to run the script

### PR DESCRIPTION
**Pull request by fsk119**
## What is the purpose of the change

*Introduce an executor to run the script.*

## Brief change log

  - *Introduce printer to print the statement and results on the console.*
  - *Introduce executor to run the statement one by one*

## Verifying this change

Please make sure both new and modified tests in this PR follow [the conventions for tests defined in our code quality guide](https://flink.apache.org/how-to-contribute/code-style-and-quality-common/#7-testing).

This change added tests and can be verified as follows:

  - *Introduce printer test to verify the style.*
  - *Introduce ScriptExecutorITCase to verify the execution results: the input contains all kinds of statements*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/application/ScriptExecutor.java
Comment (davidradl) [inline flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/application/ScriptExecutor.java:38]: This javadoc does not seem to correspond to the class name

**Comment by davidradl** [comment]
Reviewed by Chi on 28/11/24 Group to test and/or review outside of the meeting @tomncooper agreed to look at this offline

**Review by hackergin — COMMENTED** [review]
@fsk119  Thanks for your contribution, only some minor comments.

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/application/ScriptExecutor.java
Comment (fsk119) [reply flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/application/ScriptExecutor.java:170]: It is used to detect whether the statement is completed. Because Flink SQL supports `EXECUTE STATEMENT STATEMENT SET BEGIN INSERT INTO ...;`, semicolon is not the end of the statments(`END;` is the real end.). So we need to validate the statement is completed or not.

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/application/Printer.java
Comment (fsk119) [reply flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/application/Printer.java:60]: I think printer should output the content it receives. It's up to the ScriptExecutor whether keep the semicolon for each splitted statement. The `ScriptExecutor#ResultIterator` always add the current char into the buffer.

**Review by hackergin — APPROVED** [review]
@fsk119  Thanks for the update. LGTM

**Comment by davidradl** [comment]
Reviewed by Chi on 05/12/24 Need a committer to review / merge. @tomncooper expressed an interest to review

**Comment by fsk119** [comment]
@flinkbot run azure

**Comment by fsk119** [comment]
Welcome for any opinions about this PR @davidradl @tomncooper 

But please also note that we should not take too long, as I have other tasks to complete afterward.

**Comment by fsk119** [comment]
@davidradl Do you mind if I merge this PR now? If you have any other questions, we can discuss the details in the JIRA issue.

**Comment by fsk119** [comment]
Merging...

**Pull request merged**
